server/services/Authentication/signup.go

```diff
@@ -0,0 +1,104 @@
+package authentication
+
+import (
+	"fmt"
+	"main/server/db"
+	"main/server/model"
+	"main/server/request"
+	"main/server/response"
+	"main/server/services/token"
+	"main/server/utils"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+func AlreadyExists(data string) bool {
+
+	return db.RecordExist("users", data, "email")
+
+}
+
+func SignupService(ctx *gin.Context, input *request.AuthRequest) {
+
+	var user model.User
+
+	//check if the newPassword and password are the same
+	if input.NewPassword != input.Password {
+		response.ShowResponse(utils.PASSWORD_NOT_MATCH, utils.HTTP_BAD_REQUEST, utils.FAILURE, "", ctx)
+		return
+	}
+	//check the credentials if already exists
+	if AlreadyExists(input.Email) {
+
+		fmt.Println("email already exists", input.Email)
+		response.ShowResponse(utils.EMAIL_EXISTS, utils.HTTP_BAD_REQUEST, utils.FAILURE, "", ctx)
+		return
+	}
+
+	user.Email = input.Email
+	user.FullName = input.FullName
+	//encrypt the password then store in db
+
+	encryptedPassword, err := utils.HashPassword(input.Password)
```

Comment:
should it be `input.Password` or `input.NewPassword`

Reply:
Now that we have only one field of password(it should be input.Password) sir